pilot/pkg/proxy/envoy/v2/pushqueue.go

```diff
@@ -34,19 +34,56 @@ type PushEvent struct {
 	full bool
 }
 
+func (event *PushEvent) Merge(other *PushEvent) *PushEvent {
+	if event == nil {
+		return other
+	}
+	if other == nil {
+		return event
+	}
+	merged := &PushEvent{}
+	merged.push = other.push
+	merged.full = event.full || other.full
+	merged.start = event.start
+
+	// When full push, do not care about edsUpdatedServices
+	if !merged.full {
+		edsUpdates := map[string]struct{}{}
+		for endpoint := range other.edsUpdatedServices {
+			edsUpdates[endpoint] = struct{}{}
+		}
+		for endpoint := range event.edsUpdatedServices {
+			edsUpdates[endpoint] = struct{}{}
+		}
+		merged.edsUpdatedServices = edsUpdates
+	}
```

Comment:
isn't this logic in model/push_context.go as well?

Reply:
Its a different object, although it is similar.. this one was around before, push_context used to just be a `bool`.